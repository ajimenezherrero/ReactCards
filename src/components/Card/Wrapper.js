import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 30%;
  min-height: 200px;

  box-shadow: 3px 3px 5px 6px #ccc;
`;

Wrapper.propTypes = {
};

export default Wrapper;
