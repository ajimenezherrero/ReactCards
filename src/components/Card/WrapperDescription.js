import styled from 'styled-components';
import PropTypes from 'prop-types';

const WrapperDescription = styled.div`
  padding: 15px;
  flex: 1;
`;

WrapperDescription.propTypes = {
};

export default WrapperDescription;
