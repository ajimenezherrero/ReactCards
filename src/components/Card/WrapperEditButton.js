import styled from 'styled-components';
import PropTypes from 'prop-types';

const WrapperEditButton = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  padding-top: 5px;
  padding-right: 5px;
`;

WrapperEditButton.propTypes = {
};

export default WrapperEditButton;
