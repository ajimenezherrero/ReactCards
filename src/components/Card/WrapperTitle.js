import styled from 'styled-components';
import PropTypes from 'prop-types';

const getComputedStyles = ({ image }) => {
  if (image) {
    return (`
      background-image: url(${image});
    `)
  }
}

const WrapperTitle = styled.div`
  padding: 15px;
  display: flex;
  align-items: flex-end;
  font-size: 24px;
  flex: 2;

  ${props => getComputedStyles(props)}
`;

WrapperTitle.propTypes = {
};

export default WrapperTitle;
