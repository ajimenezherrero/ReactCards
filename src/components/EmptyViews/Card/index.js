import React from 'react';
import PropTypes from 'prop-types';

const CardsEmptyView = () => {
  return (
    <div>
      <label>EMPTY VIEW</label>
    </div>
  );
};

CardsEmptyView.propTypes = {
};

export default CardsEmptyView;
