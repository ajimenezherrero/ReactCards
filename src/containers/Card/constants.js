export const actionTypes = {
  ADD_CARD: 'CARD/ADD/CARD',
  SET_SHOW_ADD_CARD: 'CARD/SET/SHOW/ADD/CARD'
};
