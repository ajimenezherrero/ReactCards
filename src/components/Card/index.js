import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import EditButton from '../Buttons/CircleButton'
import Wrapper from './Wrapper';
import WrapperDescription from './WrapperDescription';
import WrapperEditButton from './WrapperEditButton';
import WrapperTitle from './WrapperTitle';

class Card extends PureComponent {
  state = {
    active: false
  }

  handleClick = () => {
    const { onEdicClick, index } = this.props;

    onEdicClick({
      index
    });
  }

  handleMouseEnter = () => {
    this.setState({
      active: true
    });
  }

  handleMouseLeave = () => {
    this.setState({
      active: false
    });
  }

  renderEditButton() {
    const { active } = this.state;

    if (active) {
      return (
        <WrapperEditButton>
          <EditButton
            onClick={this.handleClick}
            backgroundColor="green"
            size="50px"
            fontSize="12px"
          >
            Edit
          </EditButton>
        </WrapperEditButton>
      );
    }
  }

  render() {
    const { description, image, title } = this.props;

    return (
      <Wrapper onMouseLeave={this.handleMouseLeave} onMouseEnter={this.handleMouseEnter}>
        <WrapperTitle image={image}>
          {title}
        </WrapperTitle>
        <WrapperDescription>
          {description}
        </WrapperDescription>
        {this.renderEditButton()}
      </Wrapper>
    );
  }
};

Card.defaultProps = {
  title: '',
  description: '',
  image: ''
};

Card.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string
};

export default Card;
