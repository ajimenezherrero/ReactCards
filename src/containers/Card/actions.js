import { actionTypes as at } from './constants';

export const setShowAddCards = value => ({
  type: at.SET_SHOW_ADD_CARD,
  value
});

export const addCard = value => ({
  type: at.ADD_CARD,
  value
});
