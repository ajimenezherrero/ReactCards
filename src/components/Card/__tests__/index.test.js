import React from 'react';
import { shallow } from 'enzyme';


import Card from '../index';

const mockComponent = props => <Card { ...props }/>;

describe('<Card />', () => {
  it('should render component', () => {
    const tree = shallow(mockComponent());

    expect(tree).toMatchSnapshot();
  });
});
