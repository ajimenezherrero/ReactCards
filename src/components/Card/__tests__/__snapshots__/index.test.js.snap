// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    description=""
    image=""
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div
          image=""
        >
          
        </styled.div>,
        <styled.div>
          
        </styled.div>,
        undefined,
      ],
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "image": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            image=""
          >
            
          </styled.div>,
          <styled.div>
            
          </styled.div>,
          undefined,
        ],
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
            "image": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
